src/test/java/com/xceptance/common/util/CsvLineDecoderTest.java

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright (c) 2005-2023 Xceptance Software Technologies GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.xceptance.common.util;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.xceptance.xlt.api.util.XltCharBuffer;
+
+public class CsvLineDecoderTest
+{
+    void test(String s, String... expected)
+    {
+        final List<XltCharBuffer> result = CsvLineDecoder.parse(s.replace("'", "\""));
+
+        Assert.assertEquals(expected.length, result.size());
+        for (int i = 0; i < expected.length; i++)
+        {
+            Assert.assertEquals(expected[i].replace("'", "\""), result.get(i).toString());
+        }
+    }
+
+    void testException(String s, String expected)
+    {
+        try
+        {
+            CsvLineDecoder.parse(s.replace("'", "\""));
+        }
+        catch(CsvParserException e)
+        {
+            var msg = e.getMessage();
+            assertEquals(expected, msg);
+            return;
+        }
+        fail("No exception was raised");
+    }
+
+    @Test
+    public void empty()
+    {
+        var r = CsvLineDecoder.parse("");
+        assertEquals(0, r.size());
```

Comment:
```suggestion
        test("", "");
```
I would expect the decoder to return a list with a single empty field in this case. This would be in line with the justDelimiters() test farther down and also the good old CsvUtils class.

Reply:
Done.